### PR TITLE
Extend init message config params

### PR DESCRIPTION
**Pull request by CodeSandwich**
This adds functionality required by https://github.com/input-output-hk/rust-cardano/issues/610

### CODE REVIEW

File: chain-impl-mockchain/src/leadership/bft.rs [hunk 1]
```diff
@@ -105,6 +106,18 @@ impl From<PublicKey<SIGNING_ALGORITHM>> for LeaderId {
     }
 }
 
+impl Bech32 for LeaderId {
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/leadership/bft.rs]: I am not sure this is wise to start implementing this trait for LeaderId. There is no reason to believe this type and a public key are both the same object. The HRP here is not necessarily true.

File: chain-impl-mockchain/src/config.rs [hunk 2]
```diff
@@ -224,24 +287,102 @@ impl ConfigParamVariant for ConsensusVersion {
...
-        // validate that it's less than 1024 when const fns support ifs
-        Tag(tag)
+    fn from_payload(payload: &[u8]) -> Result<Self, Error> {
+        PublicKey::from_binary(payload)
+            .map(Into::into)
+            .map_err(|_| Error::SizeInvalid)
+    }
+
+    fn to_cfg_string(&self) -> String {
+        self.as_public_key().to_bech32_str()
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/config.rs]: I am not sure the encoding in `bech32` holds. Actually this comes to: why this library defines the encoding format as it should be in the `jcli`'s genesis.yaml file, which is application defined.

It seems to me the `to_cfg_string` and `from_cfg_str` functions are not needed at all here.

File: chain-impl-mockchain/src/config.rs [hunk 3]
```diff
@@ -93,84 +153,89 @@ mod serde_impl {
 
     impl<'de> Deserialize<'de> for ConfigParam {
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/config.rs]: Please, remove this module `serde_impl`. This is never compiled and will never be.

File: chain-impl-mockchain/src/config.rs (hunk 2 above)
Comment (CodeSandwich) [reply chain-impl-mockchain/src/config.rs]: I was thinking about removing `to_cfg_string` and `from_cfg_str` completely. It's only used by `ConfigParam`'s Serde implementation, which we want to get rid of anyway. I guess it's a good moment to purge it.

**Pull request merged**